### DISCUSSION TITLE
mice with() function

### DISCUSSION THREAD

**Discussion by nsdelablancac** [amices/mice#368] [Impute, analyse and pool]
Hi all! 
I'm using mice package following recommendations of Zhang here in this paper (DOI: 10.3978/j.issn.2305-5839.2015.12.63). After completing mice() step, with() function does not work with my mids output object: 
      
      #mice version: 3.8.0 2020-02-12
      with_data <- with(data = imp,
                           expr = glm(sex ~ variable1+variable2+variable3+variable4...))
      Error in `contrasts<-`(`*tmp*`, value = contr.funs[1 + isOF[nn]]) : 
      contrasts can be applied only to factors with 2 or more levels.

I checked my variable "sex" is as.factor.
with() function of mice() package does not appear in my R session, only with() from base, which I think it does not accept mids object and consequently it gives me that error. mice::with does not work neither. 
When I ask for help (?with.mids) it appears, but documentation specify with() and with.mids() does not exists.
Does anyone know why it occurs? Thank a lot!

**Comment by gerkovink** [comment]
Your model has a predictor that is a `factor`, but has fewer than 2 levels. This is an error generated in `glm` and is not related to `mice`. 

See e.g. the following reproducible example where the subset of `Species` results in a predictor with only one level. 
```r
lm(Sepal.Length ~ Sepal.Width + Species,
   data=iris[iris$Species == "setosa", ]) 
```

All the best, 

Gerko

**Comment by gerkovink** [comment]
Closing as it is not related to `mice`.

**Comment by gerkovink** [comment]
By the way: `with.mids()` is not an exported object, but can be directly accessed with `mice:::with.mids()`

**Comment by nsdelablancac** [comment]
Thanks a lot! I could solve the problem.
